This file contains to extract data from this [huge data set](https://www.kaggle.com/reddit/reddit-comments-may-2015)

![Schema](images/Sqllite_structure.png)

In [1]:
import sqlite3

In [2]:
conn = sqlite3.connect('Data/database.sqlite')

In [3]:
c = conn.cursor()

In [6]:
query = 'SELECT COUNT(*) FROM May2015'

In [7]:
c.execute(query)
all_rows = c.fetchall()
all_rows

[(54504410,)]

In [8]:
query = 'SELECT MAX(created_utc) FROM May2015'
c.execute(query)
all_rows = c.fetchall()
all_rows

[(1433116799,)]

 1433116799 is `Sunday, May 31, 2015 11:59:59 PM`

In [9]:
query = 'SELECT MIN(created_utc) FROM May2015'
c.execute(query)
all_rows = c.fetchall()
all_rows

[(1430438400,)]

1430438400 is `Friday, May 1, 2015 12:00:00 AM`

In [13]:
1433116799 + (7 * 24 * 60 * 60)

1433721599

Let's get data for one week i.e. from `1433116799` to `1433721599`, from the follwing sub-reddits.

In [26]:
subreddit_list = "('leagueoflegends', 'funny', 'dota2', 'adviceanimals', 'movies',\
                   'politics', 'anime', 'gaming', 'news', 'youshouldknow', 'worldnews',\
                   'todayilearned', 'thewalkingdead', 'lifehacks', 'pokemon', 'frugal',\
                   'starwars', 'comicbooks', 'starcraft', 'guns', 'motorcycles',\
                   'imgoingtohellforthis', 'progresspics', 'music', 'facepalm',\
                   'tf2', 'sex', 'gameofthrones', 'startrek', 'community', 'truereddit',\
                   'conspiracy', 'skyrim', 'science', 'harrypotter', 'breakingbad',\
                   'doctorwho', 'minecraft', 'libertarian', 'food', 'jokes', 'himym',\
                   'explainlikeimfive', 'conservative', 'offbeat', 'mylittlepony',\
                   'space', 'drunk', 'askscience', 'circlejerk', 'askhistorians')"
subreddit_list = ' '.join(subreddit_list.split())

In [28]:
1433116799 + 3600

1433120399

In [29]:
query = f'SELECT COUNT(*) FROM May2015 WHERE subreddit IN {subreddit_list}\
AND created_utc <= 1433120399'

c.execute(query)
all_rows = c.fetchall()
all_rows

[(3,)]

In [35]:
query = f"SELECT COUNT(*) FROM May2015 WHERE subreddit IN {subreddit_list}"
c.execute(query)
all_rows = c.fetchall()
all_rows

[(6195172,)]

In [36]:
import pandas as pd

In [38]:
query = f'SELECT * FROM May2015 WHERE subreddit IN {subreddit_list}'
df = pd.read_sql_query(query, conn)

In [39]:
df.shape

(6195172, 22)

In [41]:
df.head(3).T

0  \
created_utc                                                    1430438402   
ups                                                                     1   
subreddit_id                                                     t5_2qh22   
link_id                                                         t3_34gb0n   
name                                                           t1_cqug92d   
score_hidden                                                            0   
author_flair_css_class                                               None   
author_flair_text                                                    None   
subreddit                                                           anime   
id                                                                cqug92d   
removal_reason                                                       None   
gilded                                                                  0   
downs                                                                   0   
archived                                                                0   
author                                                            anweisz   
score                                                                   1   
retrieved_on                                                   1432703080   
body                    I can see how that could easily go. People oft...   
distinguished                                                        None   
edited                                                                  0   
controversiality                                                        0   
parent_id                                                      t1_cqufskw   

                                                                        1  \
created_utc                                                    1430438411   
ups                                                                    12   
subreddit_id                                                     t5_2qh22   
link_id                                                         t3_34gfze   
name                                                           t1_cqug99n   
score_hidden                                                            0   
author_flair_css_class                                                MAL   
author_flair_text         http://myanimelist.net/animelist/Link_of_Hyrule   
subreddit                                                           anime   
id                                                                cqug99n   
removal_reason                                                       None   
gilded                                                                  0   
downs                                                                   0   
archived                                                                0   
author                                                  The_Sordid_Critic   
score                                                                  12   
retrieved_on                                                   1432703082   
body                    I could participate in this, but by that time,...   
distinguished                                                        None   
edited                                                                  0   
controversiality                                                        0   
parent_id                                                       t3_34gfze   

                                                                        2  
created_utc                                                    1430438412  
ups                                                                     1  
subreddit_id                                                     t5_2qh22  
link_id                                                         t3_34b6gv  
name                                                           t1_cqug9aa  
score_hidden                                                            

In [42]:
df.to_feather('Data/May2015_subset.fthr')

In [45]:
df.memory_usage(deep=True).sum()

5533301288

In [46]:
f"{5533301288 / 1024**2} MB"

'5276.967323303223 MB'